stl/src/fiopen.cpp

```diff
@@ -59,12 +63,13 @@ FILE* _Xfiopen(const CharT* filename, ios_base::openmode mode, int prot) {
     mode &= ~(ios_base::ate | ios_base::_Nocreate | ios_base::_Noreplace);
 
     int n = 0;
-    while (valid[n] != 0 && valid[n] != mode) { // look for a valid mode
-        ++n;
-    }
-
-    if (valid[n] == 0) {
-        return nullptr; // no valid mode
+    for (;; ++n) { // look for a valid mode
+        if (valid[n] == mode) {
+            break;
+        }
+        if (valid[n] == 0) {
+            return nullptr; // no valid mode
+        }

```

Comment:
If we reorder these two conditions - check for `0` and `return` before checking for equality and ending the loop - then we can drop the separate check on lines 47-50. The performance of correct calls is unaffected, performance of erroneous calls potentially slowed (which we don't care about), and there's a small reduction in code size.

Reply:
Erroneous? Well, we don't know if it is erroneous, but if it is, I can fix it back